### DISCUSSION TITLE
"Vary value by Configuration" for iOS Code Signing [Scheme, Team, Profiles]??? Possible?

### DISCUSSION THREAD

**Discussion by LeoJHarris** [dotnet/maui#24621] [Q&A] [answered]
Hi All, the option is grayed out to `vary value by configuration` for iOS signing. I have an application that requires different code signing for Scheme, Team and provisioning profile depending on the config i.e. Debug, Adhoc and AppStore/Release

![image](https://github.com/user-attachments/assets/1389f516-4efc-4166-8c37-1917df5f77e1)

Is there some way to set this selectively, right now it uses the same for all configurations and that creates obvious issues with incorrect signing.

Every other config property seems to provide options for `vary value by configuration` except the one I really need. Of course I can do this manually but expected to set this as mentioned above. I would greatly appreciate assistance.

**Comment by LeoJHarris** [comment]
Anyone?

**Comment by Pastajello** [comment, marked as answer]
I guess you could. Define a custom property in your *.csproj :
`<Brand>Default</Brand>`
And then You can define different options checking this property:
```
<PropertyGroup Condition="'$(Brand)'=='Debug'">
          <CodesignKey>iPhone Distribution</CodesignKey>
	  <CodesignProvision>Automatic:AppStore</CodesignProvision>
</PropertyGroup>
```

when building you the app " dotnet publish --framework "net8.0-android" you add -p:Brand="whatever":
dotnet publish --framework "net8.0-android" -p:Brand="whatever"

**Reply by LeoJHarris** [reply]
Thanks yes I was having an issue where it was using some generic iOS config every time I updated the code signing via the GUI. Seems to be all good when setting like this method directly in csproj code editor
